src/services/gitlab.service.ts

```diff
@@ -0,0 +1,103 @@
+    const commitAuthor = this.configService.get<string>('app.commitAuthor');
+    const { name: commitAuthorName, email: commitAuthorEmail } = this.parseCommitAuthor(commitAuthor);
+
+    const body: IGitlabWebhookPushEvent['body'] = {
+      ref: this.configService.get<string>('app.branch'),
+      project: {
+        id: this.configService.get<string>('app.projectId'),
+      },
+      commits: [
+        {
+          id: this.configService.get<string>('app.commit'),
+          message: this.configService.get<string>('app.commit'),
+          title: this.configService.get<string>('app.commitTitle'),
+          timestamp: this.configService.get<string>('app.commitTimestamp'),
+          author: {
+            name: commitAuthorName,
+            email: commitAuthorEmail,
+          },
+        },
+      ],
+    };
+
+    return { body, headers };
+  }
+
+  buildWebhookMergeRequestEvent(): IGitlabWebhookMergeRequestEvent {
+    const headers: IGitlabWebhookMergeRequestEvent['headers'] = {
+      'x-gitlab-event': 'Merge Request Hook',
+      'x-gitlab-event-uuid': uuidv4(),
+    };
+
+    const mergeRequestIid = this.configService.get<string>('app.mergeRequestIid');
+    const mergeRequestProjectUrl = this.configService.get<string>('app.mergeRequestProjectUrl');
+    const mergeRequestUrl = `${mergeRequestProjectUrl}/-/merge_requests/${mergeRequestIid}`;
+
+    const commitAuthor = this.configService.get<string>('app.commitAuthor');
+    const { name: commitAuthorName, email: commitAuthorEmail } = this.parseCommitAuthor(commitAuthor);
+
+    const body: IGitlabWebhookMergeRequestEvent['body'] = {
+      user: {
+        name: commitAuthorName,
+      },
+      project: {
+        id: this.configService.get<string>('app.projectId'),
+      },
+      object_attributes: {
+        created_at: this.configService.get<string>('app.mergeRequestCreatedAt'),
+        id: this.configService.get<string>('app.mergeRequestId'),
+        iid: this.configService.get<string>('app.mergeRequestIid'),
+        source_branch: this.configService.get<string>('app.mergeRequestSourceBranch'),
+        target_branch: this.configService.get<string>('app.mergeRequestTargetBranch'),
+        title: this.configService.get<string>('app.mergeRequestTitle'),
+        updated_at: this.configService.get<string>('app.mergeRequestUpdatedAt'),
+        last_commit: {
+          id: this.configService.get<string>('app.commit'),
+          message: this.configService.get<string>('app.commitMessage'),
+          title: this.configService.get<string>('app.commitTitle'),
+          timestamp: this.configService.get<string>('app.commitTimestamp'),
+          author: {
+            name: commitAuthorName,
+            email: commitAuthorEmail,
+          },
+        },
+        state: this.configService.get<string>('app.mergeRequestState'),
+        url: mergeRequestUrl,
+        // TODO: get actual "action" value. Can be calculated based on merge request state
+        action: 'update',
```

Comment:
`action` is specific field to webhook event, but in theory it can be calculated from [Resource state events API](https://docs.gitlab.com/ee/api/resource_state_events.html#list-project-merge-request-state-events) as latest merge request state event